core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,214 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for solution.
+ */
+
+oppia.controller('StateSolution', [
+  '$scope', '$rootScope', '$modal', 'editorContextService', 'alertsService',
+  'INTERACTION_SPECS', 'stateSolutionService', 'explorationStatesService',
+  'SolutionVerificationService', 'oppiaExplorationHtmlFormatterService',
+  'stateInteractionIdService', 'stateHintsService', 'UrlInterpolationService',
+  'SolutionObjectFactory', 'responsesService', 'explorationContextService',
+  'explorationWarningsService', 'INFO_MESSAGE_SOLUTION_IS_INVALID',
+  function(
+    $scope, $rootScope, $modal, editorContextService, alertsService,
+    INTERACTION_SPECS, stateSolutionService, explorationStatesService,
+    SolutionVerificationService, oppiaExplorationHtmlFormatterService,
+    stateInteractionIdService, stateHintsService, UrlInterpolationService,
+    SolutionObjectFactory, responsesService, explorationContextService,
+    explorationWarningsService, INFO_MESSAGE_SOLUTION_IS_INVALID) {
+    $scope.stateSolutionService = stateSolutionService;
+    $scope.inlineSolutionEditorIsActive = false;
+    $scope.correctAnswerEditorHtml = '';
+    $scope.SOLUTION_EDITOR_FOCUS_LABEL = (
+      'currentCorrectAnswerEditorHtmlForSolutionEditor');
+    explorationWarningsService.updateWarnings();
+    $scope.stateHintsService = stateHintsService;
+    $scope.correctAnswer = null;
+
+    $scope.isSolutionValid = function() {
+      return explorationStatesService.isSolutionValid(
+        editorContextService.getActiveStateName());
+    };
+
+    $scope.inlineSolutionEditorIsActive = false;
```

Comment:
This duplicates line 34. Please put all the initialization code in one place and consider alphabetizing (or some other reasonable grouping/ordering) to avoid issues like this.